anyway/parsers/infographics_data_cache_updater.py

```diff
@@ -106,6 +107,43 @@ def get_infographics_data_from_cache_by_road_segment(road_segment_id, years_ago)
         return {}
 
 
+def get_cache_retrieval_query(params: RequestParams):
+    res = params.resolution
+    if res == BE_CONST.ResolutionCategories.SUBURBAN_ROAD:
+        return (
+            db.session.query(InfographicsRoadSegmentsDataCache)
+            .filter(
+                InfographicsRoadSegmentsDataCache.road_segment_id
+                == int(params.location_info["road_segment_id"])
+            )
+            .filter(InfographicsRoadSegmentsDataCache.years_ago == int(params.years_ago))
+        )
+    else:
+        msg = f"Cache unsupported resolution: {res}, params:{params}"
+        logging.error(msg)
+        raise ValueError(msg)
+
+
+def get_infographics_data_from_cache_by_location(request_params: RequestParams) -> Dict:
+    logging.error("get_infographics_data_from_cache_by_location: not yet implemented")
+    query = get_cache_retrieval_query(request_params)
+    db_item = query.first()
+    logging.debug(f"retrieved from cache {type(db_item)}:{db_item}"[:70])
```

Comment:
I am not sure - `type(db_item)` will return the same value in all cases?
Why there is `[:70]` in the end of the line? - do you want to truncate the text?

Reply:
It should always return same type. At the beginning I did not put the right type, so I had a bug there. Now it can be removed. The print is truncated to avoid long print. The purpose of the print is to verify that it is a valid infographics data output, so the beginning suffice.